src/Files.App/Utils/Storage/Enumerators/Win32StorageEnumerator.cs

```diff
@@ -261,6 +262,15 @@ CancellationToken cancellationToken
 				itemType = itemFileExtension.Trim('.') + " " + itemType;
 			}
 
+			int imageHeight = 0, imageWidth = 0;
+			if (FileExtensionHelpers.IsImageFile(itemFileExtension))
```

Comment:
I'm wondering if we need to initialize this data when enumerating items. It might be better for performance if we retrieve it on demand from the getter in ListedItem. @hishitetsu

Reply:
My thoughts are that at this stage in the pipeline its already doing a lot of metadata gathering anyway, after doing some research I went with the FileStream approach as well as Image.FromFile gets a bad rep

Reply:
I think we can get dimensions from `System.Image.Dimensions` property. I don't know which one performs better though.

Reply:
@btomblinson

Reply:
That is how `PropertiesInformation.json` gets it, which I looked into but couldn't follow it so found it too complex for usage in this scenario, the approach I took uses standard .NET APIs that are more understandable, I also found `System.Drawing.Bitmap` which might be worth a look, do we have any suggestions on how to benchmark the 3 approaches?

Reply:
I tested with a 21MB image for performance. Screenshots of code and results, a copy of the .cs and a copy of the image are attached. Consistently, using `System.Drawing.Image` has clocked in at 5ms or less and the `System.Drawing.Bitmap` approach took 20+ ms. I do not know how to test `System.Image.Dimensions` since thats win32 api but I believe I've shown my usage of  `System.Drawing.Image` is more than sufficient for this use case and might even confirm its the best approach period.

[Performance.zip](https://github.com/files-community/Files/files/14717849/Performance.zip)

Reply:
> My thoughts are that at this stage in the pipeline its already doing a lot of metadata gathering anyway, after doing some research I went with the FileStream approach as well as `Image.FromFile` gets a bad rep

The below ZIP confirms my research that FileStream is definitely the fastest option

[Performance.zip](https://github.com/files-community/Files/files/14717937/Performance.zip)

Reply:
In that case let's stick with FileStream, but we can optimize further by only fetching the data when needed. True, most other properties are loaded right away, but it reduces performance and it's something we want to optimize moving forward.

Reply:
What if anything do you need me to do?